refactor(ast): rename the `visited_node` marker to `ast`.

rzvxa opened the pull request:
closes #4282
I went with `#[ast(visit)]` but we can change it to 2 separate markers as suggested by @overlookmotel in the issue.

rzvxa commented:
* **#4289** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4289?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rzvxa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented on crates/oxc_ast/src/ast_impl/js.rs:
```diff
@@ -1,6 +1,3 @@
-// NB: `#[visited_node]` attribute on AST nodes does not do anything to the code in this file.
```
Can we keep these comments, but just change `#[visited_node]` to `#[ast]`? The intent is for `#[ast]` to become a real macro in future, but it isn't right now - and anyway when it does we'll need to have a comment here explaining what it does.

rzvxa replied:
These are removed from the implementation files where we don't have any occurrence of the `visited_node`(and now `ast`). I can move them to a separate PR if you'd prefer. That's just a cleanup I've missed when separating ast files into x.rs and x_impl.rs

overlookmotel replied:
Oh I'm sorry, my mistake. I wasn't paying attention to which files these comments were in. No need to put in a separate PR. Just ignore me!

overlookmotel commented:
@rzvxa Is this stack ready for review? (or at least bottom 3 PRs of it?)

rzvxa commented:
> @rzvxa Is this stack ready for review? (or at least bottom 3 PRs of it?) Or you still working on it?

Yes, the bottom 3 are ready to be reviewed.

rzvxa committed:
refactor(ast): rename the `visited_node` marker to `ast`. (#4289)

closes #4282
I went with `#[ast(visit)]` but we can change it to 2 separate markers as suggested by @overlookmotel in the issue.
